controller/scheduler/fleet.py

```diff
@@ -112,6 +113,18 @@ def _create_container(self, name, image, command, unit, **kwargs):
             l.update({'cpu': '-c {}'.format(cpu)})
         else:
             l.update({'cpu': ''})
+        # set unit hostname
+        hostname = settings.UNIT_HOSTNAME
+        if hostname == "default":
+            l.update({'hostname': ''})
+        elif hostname == "application":
+            # replace underscore with dots, since underscore is not valid in DNS hostnames
+            dns_name = name.replace("_", ".")
```

Comment:
I considered hyphens, and I considered stripping it out entierly. I can't see how that could cause issues with some applications, and also it would make filtering easier as it will be easier to write filters (when searching through logs which includes the hostname) where `my-application.*.web.*` is found.